src/Microsoft.AspNet.Authentication.Cookies/Events/CookieAuthenticationEvents.cs

```diff
@@ -42,6 +43,28 @@ public class CookieAuthenticationEvents : ICookieAuthenticationEvents
             return Task.FromResult(0);
         };
 
+        private static bool IsAjaxRequest(HttpRequest request)
+        {		
+            var query = request.Query;		
+            if (query != null)		
+            {		
+                if (query["X-Requested-With"] == "XMLHttpRequest")		
```

Comment:
These null checks shouldn't be needed. Worse case you can combine them into a single if.